File: app/src/main/java/com/pr0gramm/app/ui/fragments/NoopOnScrollListener.java
package com.pr0gramm.app.ui.fragments;

import android.support.v7.widget.RecyclerView;

/**
 * Scroll listener that does nothing
 */
class NoopOnScrollListener extends RecyclerView.OnScrollListener {
    @Override
    public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
        // empty
    }

    @Override
    public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
        // do nothing
    }
}

File: app/src/main/java/com/pr0gramm/app/ui/fragments/NoopOnScrollListener.kt
package com.pr0gramm.app.ui.fragments

/**
 * Scroll listener that does nothing
 */
internal class NoopOnScrollListener : androidx.recyclerview.widget.RecyclerView.OnScrollListener() {
    override fun onScrollStateChanged(recyclerView: androidx.recyclerview.widget.RecyclerView, newState: Int) {
        // empty
    }

    override fun onScrolled(recyclerView: androidx.recyclerview.widget.RecyclerView, dx: Int, dy: Int) {
        // do nothing
    }
}
